src/replication/DealReplicationWorker.ts

```diff
@@ -55,10 +58,34 @@ export default class DealReplicationWorker extends BaseService {
 
   private async pollWork (): Promise<boolean> {
     this.logger.debug(`${this.workerId} - Polling for work`);
+    await this.checkCronChange();
     const hasDoneWork = await this.pollReplicationWork();
     return hasDoneWork;
   }
 
+  private async checkCronChange (): Promise<void> {
+    const activeCronWorks = await Datastore.ReplicationRequestModel.find({
+      cronSchedule: { $ne: null },
+      workerId: { $ne: null },
+      status: 'active'
+    });
+
+    for (let i = 0; i < activeCronWorks.length; i++) {
+      const request2Check = activeCronWorks[i];
+      if (this.cronRefArray.has(request2Check.id) && this.cronRefArray.get(request2Check.id)![0] !== request2Check.cronSchedule) {
```

Comment:
Can you use https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements
So we get rid of the magic numbers